src/lib.rs

```diff
@@ -155,9 +158,11 @@ pub trait SynthLanguage: egg::Language + Send + Sync + Display + FromOp + 'stati
         let mut rename_node = |node: &Self| match node.to_var() {
             Some(sym) => {
                 let len = map.len();
-                let var = map
-                    .entry(sym)
-                    .or_insert_with(|| format!("?{}", letter(len)).parse().unwrap());
+                let var = map.entry(sym).or_insert_with(|| {
+                    format!("?{}{}", node.get_type(), letter(len))
```

Comment:
so this makes sure that alpha-renaming preserves the type information, is that right?

Reply:
Yeah, the variables will be <type-prefix><alpha-renamed symbol>